### ISSUE TITLE
Database error

### ISSUE THREAD

**Issue by mrlioncub** [lirantal/daloradius#58]
Install from https://github.com/lirantal/daloradius/archive/master.tar.gz (29.06.2018)
PHP Version 7.0.30
daloRADIUS -> Reports Page ->  Last Connection Attempts:

```
Database error
Error Message: DB Error: syntax error
Debug info: SELECT radpostauth. FROM radpostauth WHERE (radpostauth. LIKE '%%') AND ( >='2018-06-01' AND <='2018-07-01') [nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'FROM radpostauth WHERE (radpostauth. LIKE '%%') AND ( >='2018-06-01' AND <='2' at line 2]

Fatal error: Call to undefined method DB_Error::numRows() in /usr/local/lib/php/DB.php on line 985
```

Remove User:
```
> Database error
> Error Message: DB Error: syntax error
> Debug info: DELETE FROM radpostauth WHERE ='user' [nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '='user'' at line 1]
```

Login Daloradius with cyrilic password:
```
Database error
Error Message: DB Error: unknown error
Debug info: SELECT id, username FROM operators WHERE username = 'administrator' AND password = 'кфвшгы' [nativecode=1267 ** Illegal mix of collations (latin1_swedish_ci,IMPLICIT) and (utf8_unicode_ci,COERCIBLE) for operation '=']

Fatal error: Call to undefined method DB_Error::numRows() in /usr/local/lib/php/DB.php on line 985
```

**Comment by alimp5** [comment]
Greetings!

I downloaded the last version (1.0.0)
i got same error on `General Reports' for Last-Connection-Attempts :

```

Database error
Error Message: DB Error: syntax error
Debug info: SELECT radpostauth. FROM radpostauth WHERE (radpostauth. LIKE '%') AND ( >='2019-04-01' AND <='2019-04-30') [nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'FROM radpostauth WHERE (radpostauth. LIKE '%') AND ( >='2019-04-01' AND <='20' at line 2]
```

**Comment by MiguelVis** [comment]
Hi @alimp5 

I guess I have discovered a potential bug in `rep-lastconnect.php` after checking your issue:

```
	if (isset($configValues['FREERADIUS_VERSION']) && ($configValues['FREERADIUS_VERSION'] == '2')) {
		$tableSetting['postauth']['user'] = 'username';
		$tableSetting['postauth']['date'] = 'authdate';
	} elseif (isset($configValues['FREERADIUS_VERSION']) && ($configValues['FREERADIUS_VERSION'] == '1')) {
		$tableSetting['postauth']['user'] = 'user';
		$tableSetting['postauth']['date'] = 'date';
	}
```

I wonder you are running daloRADIUS under FreeRadius 3 and you have set `$configValues['FREERADIUS_VERSION']` to the value `'3'`.

If that is the case, both `$tableSetting['postauth']['user']` and `$tableSetting['postauth']['date']` will be undefined, so the SQL queries would be bad formed:

```
$sql = "SELECT ".
			$configValues['CONFIG_DB_TBL_RADPOSTAUTH'].".".$tableSetting['postauth']['user']." 
		FROM ".$configValues['CONFIG_DB_TBL_RADPOSTAUTH']. 
        " WHERE (".$configValues['CONFIG_DB_TBL_RADPOSTAUTH'].".".$tableSetting['postauth']['user']." LIKE '".
	$dbSocket->escapeSimple($usernameLastConnect)."%') $radiusReplySQL ".
		" AND (".$tableSetting['postauth']['date']." >='$startdate' AND ".$tableSetting['postauth']['date']." <='$enddate') ";
```

I have prepared and merged to master branch the PR #99 that possibly fix your issue.

Please, check and confirm it that's the case.

**Comment by MiguelVis** [comment]
No reply has arrived and I understand that PR #99 fix this issue. So, closed.
